app/src/main/java/com/calendar/MainActivity.java

```diff
@@ -104,6 +138,8 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         switch (item.getItemId()) {
+            case android.R.id.home:
+                return drawerToggle != null && drawerToggle.onOptionsItemSelected(item) || super.onOptionsItemSelected(item);
```

Comment:
I would propose to move this return statement to new method. To better explain its meaning.

Reply:
Moved to method.